esp/services/ws_config2/ws_config2Service.cpp

```diff
@@ -657,22 +661,27 @@ void Cws_config2Ex::getNodeResponse(const std::shared_ptr<EnvironmentNode> &pNod
         pInsertInfo->setCategory(pSchemaItem->getProperty("category").c_str());
         pInsertInfo->setRequired(pSchemaItem->isRequired());
         pInsertInfo->setTooltip(pSchemaItem->getProperty("tooltip").c_str());
-        std::string limitType = pSchemaItem->getProperty("insertLimitType");
-        if (!limitType.empty())
+        if (it->m_limitChoices)
         {
+            addItem = false;   // don't add this item unless we find a limited choice
             pInsertInfo->setFixedChoices(true);
             IArrayOf<IEspChoiceLimitType> fixedChoices;
-            for (auto &&fc : (*it).m_itemLimits)
+            for (auto &fc : (*it).m_itemLimits)
             {
                 Owned<IEspChoiceLimitType> pChoice = createChoiceLimitType();
                 pChoice->setDisplayName(fc.itemName.c_str());
                 std::string itemType = pSchemaItem->getItemType() + "@" + fc.attributeName + "=" + fc.attributeValue;
                 pChoice->setItemType(itemType.c_str());
                 fixedChoices.append(*pChoice.getLink());
+                addItem = true;
```

Comment:
why not m_limitChoices.ordinality() != 0 in line 681 instead of keeping this bool and setting on every iteration?
I thikn IArrayOf has an ordinality()

Reply:
m_limitChoices is a bool. It indicates that this insertable item has limits. Because it has limits, a list of choices is built. Then, only if there were choices found is the item added to the list. I did make a change to set the value of addItem based on the ordinality of fixedChoices which reduces the set of addItem = true on each pass.

Reply:
I'm glad you noticed I was actually referring to the IArrayOf "fixedChoices". Thanks for the explanation of the boolean that hopefully goes away based on my observation.